Focus and LaTeX Matrix Changes

danielshanmore opened the pull request:
Moves browser focus to appropriate text field after button actions and
on page load.  Matrix entry now accepts the body of a LaTeX format
matrix (without \begin or \end).

danielshanmore committed:
Focus and LaTeX Matrix Changes

Moves browser focus to appropriate text field after button actions and
on page load.  Matrix entry now accepts the body of a LaTeX format
matrix (without \begin or \end).

pstaabp commented on matrix/gauss-elim.js:
```diff
@@ -80,8 +82,14 @@ function storeMatrix()
+            $j("#horizLine").attr("checked", settings.horizLine);
+        }
+        // Same filter as before, but additionally handles \\, \hline, and & surrounded by any number of spaces. 
+        matrices[0] = new Matrix($j("#matrix-entry").val().replace(/\s*&\s*/g, " ").replace(/\\\\/g, "").replace(/\n/g, ";").replace(/;*$/, "").replace(/\\hline/g, ""));
```
This should be put into the matrix-operations.js file.  Probably right at the top.

pstaabp commented on matrix/gauss-elim.js:
```diff
@@ -12,7 +12,7 @@ $j(document).ready(function()
               
         $j("#settings").css("display","none");
-        $j("#clear-matrix-button").click(function () { $j("#matrix-entry").val('');} ); 
+        $j("#clear-matrix-button").click(function () { $j("#matrix-entry").val(''); $j("#matrix-entry").focus();} ); 
```
You can also do this a little shorter with
```
$j("#clear-matrix-button").click(function () { $j("#matrix-entry").val('').focus();} );
```

pstaabp commented:
Overall, this looks great.  I put a couple of comments on the file itself. One just shortens one line and another puts the parsing into a different file so it can be used in other places.

danielshanmore committed:
Clickable Pivots, more LaTeX and Focus

The pivots are clickable after clicking the Select Pivot button, and
doing so auto fills the input box and highlights the selection.  All the
string filtering for Matrix creation was moved to the constructor, so
both pages can enter matrices in LaTeX mode.  More focus changes to the
Matrix Operations page.

pstaabp commented on matrix/gauss-elim.js:
```diff
@@ -315,5 +319,64 @@ function undo(obj)
+// Called after piv-button is clicked, makes the elements of the currently typeset matrix clickable
+function clickablePivots() {
+    // This sections grabs the current matrix and stores its name and elements
+    var outCount = document.getElementById("output").childElementCount;
```
You could do a bit of jquery here instead.  For example, this line could be
```
var outCount = $("#output").children().length

pstaabp commented:
This is looking good.  I made one comment that you could do a little bit of jquery if you want, but not necessary, but it's a bit funny to mix both jquery and non-jquery code.  

A few other thoughts:
1. Can you make the last matrix have a different cursor icon when you're over a cell to pivot on.  [Here's](https://www.w3schools.com/csSref/pr_class_cursor.asp) a list of default icons.  I'm think either cell or crosshair. The best way to do this is to give the elements a class like "pivotable" and then in the file `matrix-ops.css` add a line to switch the cursor. 
2. It seems a bit more natural to click on the cell and then just do the pivot instead of having to do the extra step of clicking the pivot.
3. I'd like to make this an option that can be selected in the settings.  Take a look at the latex button as an example.  
4. If the matrix is not all integers, then piv doesn't work.  I need to have a better error message for this.  You should then do a pivot instead, which make the pivot element a 1.  I think this needs to be stored in the Matrix object somehow.  I'll think about doing this.

danielshanmore committed:
Merge remote-tracking branch 'refs/remotes/pstaabp/master' into Focus-and-LaTeX-Matrix-changes

danielshanmore committed:
Pivot Changes, Absolute Value Function for Rationals

Changed most of the code to jQuery, pivotable entries now change cursor
and color on hover.  Removed the previously added global variables.
Tried to provide the Rational object with an absolute value function.

danielshanmore committed:
jQuery Cleanup, Pivot Selection toggles from settings, Pivot on 0 bugfix.

Select Pivot button must now be enabled first from settings menu.  An
attempt to pivot on 0 no longer makes the next pivot happen twice.

pstaabp commented:
Looks good.  I'm going to pull this in.
